### PR TITLE
Server: Add Normalization on Admin APIs

### LINKED ISSUE #14056: UI: fail to show table information if the table name is upper case
Please describe the issue you observed:

- What did you do?
1. Execute the following SQL:
  CREATE database test1;
  SET database = test1;
  CREATE TABLE T1 (k INT PRIMARY KEY);
  CREATE TABLE t2 (k INT PRIMARY KEY);
2. In the admin UI
  In "DATABASES" tab, database "test1", click table "T1".

- What did you expect to see?
The information of the table "T1" including the DDL to create this table.

- What did you see instead?
"No results."

### PR DESCRIPTION
**Pull request by xphoniex**
Fixes #14056

Table names are transformed into lower case so we need to normalize table name when looking for them on API endpoints.

### CODE REVIEW

**Comment by tamird** [comment]
Thanks for the PR! I think we need to add a test. Assigned @mrtracy to shepherd.

**Comment by mrtracy** [comment]
This is easily reproduced with a test case added to `pkg/storage/admin_test.go`:

```go
func TestAdminAPITableDetailsUppercase(t *testing.T) {
	defer leaktest.AfterTest(t)()
	testAdminAPITableDetailsInner(t, "TEST", "TBL")
}
```

This can be added immediately after the TestAdminAPITableDetails test - perhaps add a comment inside the function that it is a regression of #14056.

**Comment by xphoniex** [comment]
@mrtracy good to go.

File: pkg/server/admin_test.go [hunk 1]
```diff
@@ -400,6 +400,12 @@ func TestAdminAPITableDetails(t *testing.T) {
 	testAdminAPITableDetailsInner(t, "test", "tbl")
 }
 
+// #14056
```
Comment (mrtracy) [inline pkg/server/admin_test.go]: I don't think this is going to pass our linter - function comments in go need to be of the form `// [ObjectBeingCommented] [description]`, or must not be present at all.  In this case, that would mean the comment has to be of a form similar to `// TestAdminAPITableDetailsUppercase ensures the Admin API works correctly with uppercase table names. This is a regression of #14056`.

If you don't want to be quite so verbose, you could also just put the comment inside the body of the function.  However, even then the comment should be slightly more descriptive like "// Regression test for issue #14056", the issue number without context is not that helpful - it needs to be distinguished from an active issue or pull request.

Comment (tamird) [inline pkg/server/admin_test.go]: These lint rules don't apply to test functions.

If you are feeling altruistic, you could convert this test and its friends to sub-tests:

```golang
func TestAdminAPITableDetails(t *testing.T) {
  defer leaktest.AfterTest(t)()

  for _, tc := range []struct{
    name, dbName, tableName string
  }{
   { name: "lower", dbName: "test", tableName: "tbl" },
   { name: "lower with space", dbName: "test test", tableName: "tbl tbl" },
   { name: "upper", dbName: "TEST", tableName: "TBL" },
  }{
    t.Run(tc.name, func(t *testing.T) {
      // body of testAdminAPITableDetailsInner
    })
  }
}
```

**Comment by xphoniex** [comment]
does this have anything to do with my commit?

```
[TestLogic/default/scale] logic_test.go:1580: 
	testdata/logic_test/scale:9: expected success, but found
	pq: storage/store.go:2485: rejecting command with timestamp in the future: 1492028648225164299 (540.407642ms ahead)
logic_test.go:1590: 
	testdata/logic_test/scale:9: error in statement, skipping to next file
```

**Comment by tamird** [comment]
It does not. I'd rebuild the tests for you but since changes have been requested anyway, that failure should clear up when you amend and push.

**Comment by mrtracy** [comment]
That is a type of error that can occasionally occur on the VMs we use for CI tests when their clocks are adjusted during a test.  A future timestamp like that won't result in any consistency error in Cockroach (we detect it and throw an error), but unfortunately this will often break some random test's expectation that some operation would succeed without error, resulting in an unexpected test failure.

**Comment by xphoniex** [comment]
@tamird @mrtracy , hey guys, who should approve this?

**Comment by tamird** [comment]
LGTM

**Pull request merged**